### PR TITLE
MAT-6866: Implement special characters export check

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-6866](https://jira.cms.gov/browse/MAT-6866)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/testCaseLanding/qdm/TestCaseList.tsx [hunk 1]
```diff
@@ -41,13 +41,13 @@ import {
...
 import { uniqWith } from "lodash";
-
+import checkSpecialCharacters from "../common/checkSpecialCharacters";
 export const IMPORT_ERROR =
   "An error occurred while importing your test cases. Please try again, or reach out to the Help Desk.";
-
 export const coverageHeaderRegex =
   /<h2> .* Clause Coverage: (\d*\.\d+|\d*|NaN)%<\/h2>/i;
-
+export const EXPORT_ERROR_CHARACTERS =
```
Comment (adongare) [inline src/components/testCaseLanding/qdm/TestCaseList.tsx]: when I read this variable name, I expected a set of characters. could we change the name to mabe special_character_error_message or something similar? also, fullstop or comma between the words "exported" and "some"

File: src/components/testCaseLanding/common/checkSpecialCharacters.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { EXPORT_ERROR_CHARACTERS } from "../qdm/TestCaseList";
+const checkSpecialCharacters = (testCases: TestCase[]): string[] => {
+  const specialChars = /[`!@#$%^&*()_\+=\[\]{};':"\\|,.<>\/?~]/;
+  let series, title;
+  const failedTCs = [];
+  testCases?.forEach((tc) => {
+    series = specialChars.test(tc.series);
+    title = specialChars.test(tc.title);
+    if (series || title) {
+      failedTCs.push(`${EXPORT_ERROR_CHARACTERS}${tc.series} ${tc.title}`);
```
Comment (adongare) [inline src/components/testCaseLanding/common/checkSpecialCharacters.ts:11]:
```suggestion
      failedTCs.push(`${EXPORT_ERROR_CHARACTERS} ${tc.series} ${tc.title}`);
```

File: src/components/testCaseLanding/qdm/TestCaseList.tsx [hunk 3]
```diff
@@ -529,7 +538,7 @@ const TestCaseList = (props: TestCaseListProps) => {
                 onDeleteAllTestCases={() =>
                   setOpenDeleteAllTestCasesDialog(true)
                 }
-                onExportQRDA={() => exportQRDA()}
+                onExportQRDA={checkTestCases}
```
Comment (adongare) [inline src/components/testCaseLanding/qdm/TestCaseList.tsx]: I think the primary action here is to export QRDA with some validations. Maybe calling this function from `exportQRDA` would make it more clear.

**Review by adongare — APPROVED** [review]
